packages/mui-joy/src/Modal/Modal.tsx

```diff
@@ -117,6 +117,8 @@ const Modal = React.forwardRef(function ModalU(inProps, ref) {
     component,
     slots = {},
     slotProps = {},
+    // @ts-ignore TODO: Add it in the types - how can we make the component extendable?
+    ownerState: ownerStateProp,
```

Comment:
This is one things that we will need to resolve if we decide to go with component composition approach.